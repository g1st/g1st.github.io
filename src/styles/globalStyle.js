import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
 
  :root {
    font-family: 'Open Sans', Arial, Helvetica, sans-serif;
  }
  
  body {
    transition: all 0.25s linear;
    color: ${({ theme }) => theme.colors.text};
    background: ${({ theme }) => theme.colors.primary};
  }

  p {
    line-height: 1.15;
  }
`;
