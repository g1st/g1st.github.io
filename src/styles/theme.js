export const dark = {
  name: "dark",
  colors: {
    primary: "#31263E",
    secondary: "#44355B",
    tertiary: "#C42021",
    text: "#ffffff",
    themeToggle: "#ffffff",
    footer: "#000000",
    danger: "#f32013",
  },
};

export const light = {
  name: "light",
  colors: {
    primary: "#c5d5c5",
    secondary: "#f0f0f0",
    tertiary: "#9fa9a3",
    text: "#2e2e2e",
    themeToggle: "#2e2e2e",
    footer: "#9fa9a3",
    danger: "#c91d12",
  },
};
