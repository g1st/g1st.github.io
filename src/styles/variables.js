export const breakpoints = {
  xs: "600px",
  sm: "768px",
  md: "992px",
  lg: "1200px",
};

export const sizes = {
  headerHeight: "6rem",
};
