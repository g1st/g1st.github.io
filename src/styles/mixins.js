import styled, { css } from "styled-components";
import { breakpoints } from "./variables";

export const media = Object.keys(breakpoints).reduce((accumulator, label) => {
  accumulator[label] = (...args) => css`
    @media screen and (min-width: ${breakpoints[label]}) {
      ${css(...args)};
    }
  `;
  return accumulator;
}, {});

export const Button = styled.button`
  font-size: 1rem;
  padding: 0.5rem 1.25rem;
  border: 2px solid ${({ theme }) => theme.colors.tertiary};
  border-radius: 3px;
  background: transparent;
  color: ${({ theme }) => theme.colors.text};
  letter-spacing: 2px;
  text-transform: uppercase;
  transition: all 0.1s ease-in-out;
  font-weight: 600;

  :hover {
    cursor: pointer;
    background: ${({ theme }) => theme.colors.tertiary};
  }
`;

export const AnchorLink = styled.a`
  font-size: 1rem;
  padding: 0.5rem 1.25rem;
  border: 2px solid ${({ theme }) => theme.colors.tertiary};
  border-radius: 3px;
  background: transparent;
  color: ${({ theme }) => theme.colors.text};
  letter-spacing: 2px;
  text-transform: uppercase;
  transition: all 0.1s ease-in-out;
  font-weight: 600;

  :hover {
    cursor: pointer;
    background: ${({ theme }) => theme.colors.tertiary};
  }
`;
