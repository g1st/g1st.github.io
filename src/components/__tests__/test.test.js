import React from "react";
import styled from "styled-components";
import { render } from "@testing-library/react";
import { dark } from "../../styles/theme";

const Button = styled.button`
  color: ${(props) => props.theme.colors.primary};
`;

const Title = () => <Button theme={dark}>Tests are working</Button>;

test("Tests works", () => {
  const { getByText } = render(<Title />);
  const button = getByText(/working/i);

  expect(button).toHaveTextContent("Tests are working");
});

test("Works with styled component", () => {
  const { container } = render(<Title />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule("color", dark.colors.primary);
});
