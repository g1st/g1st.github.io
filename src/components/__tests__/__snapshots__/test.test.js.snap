// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works with styled component 1`] = `
.c0 {
  color: #282C35;
}

<button
  class="c0"
>
  Tests are working
</button>
`;
